spec/support/context/with_certificate.rb

```diff
@@ -1,11 +1,18 @@
+require 'spec_helper_acceptance'
+
 shared_context 'with certificate copied to system under test' do
   before(:all) do
     @client = AzureHelper.new
     @template = 'azure_vm_classic.pp.tmpl'
 
     @local_private_key_path = File.join(Dir.getwd, 'spec', 'acceptance', 'fixtures', 'insecure_private_key.pem')
+
     @remote_private_key_path = '/tmp/id_rsa'
 
+    if ENV['PUPPET_AZURE_BEAKER_MODE'] =~ /^apply|^agent/ && default['platform'] =~ /^windows/
```

Comment:
If memory serves `agent` is the default, so this would do the wrong thing unless the ENV is explicit. Something like the following might be clearer:

```ruby
@remote_private_key_path = if ENV['PUPPET_AZURE_BEAKER_MODE'] !~ /^local$/ && default['platform'] =~ /^windows/
  'c:\\cygwin64\\tmp\\id_rsa'
else 
  '/tmp/id_rsa'
end
```